common/database/filter.go

```diff
@@ -27,6 +36,66 @@ func GetFieldTypes(model interface{}) map[string]string {
 	return expected_types
 }
 
+func UpdateQuerySelectorInt64(qs QuerySelector, query_type QueryType, cast_values []int64) (QuerySelector, error) {
+	switch query_type {
+	case LessThan:
+		qs["$lt"] = cast_values[0]
+	case In:
+		qs["$in"] = cast_values
+	case GreaterThan:
+		qs["$gt"] = cast_values[0]
+	case NotIn:
+		qs["$nin"] = cast_values
+	default:
+		return nil, errors.New("Invalid operation on integers")
+	}
+	return qs, nil
+}
+
+func UpdateQuerySelectorString(qs QuerySelector, query_type QueryType, cast_values []string) (QuerySelector, error) {
+	switch query_type {
+	case LessThan:
+		qs["$lt"] = cast_values[0]
+	case In:
+		qs["$in"] = cast_values
+	case GreaterThan:
+		qs["$gt"] = cast_values[0]
+	case NotIn:
+		qs["$nin"] = cast_values
+	default:
+		return nil, errors.New("Invalid operation on strings")
+	}
+	return qs, nil
+}
+
+func UpdateQuerySelectorBool(qs QuerySelector, query_type QueryType, cast_values []bool) (QuerySelector, error) {
+	switch query_type {
+	case In:
+		qs["$in"] = cast_values
+	case NotIn:
+		qs["$nin"] = cast_values
+	default:
+		return nil, errors.New("Invalid operation on booleans")
+	}
+	return qs, nil
+}
+
+func ParseQueryType(key string) (QueryType, string) {
+	query_type := In
+
+	if strings.HasSuffix(key, "Lt") {
+		key = key[0 : len(key)-2]
```

Comment:
Is it possible for `len(key)` to be less than 2? (Or 3 below?)

Reply:
If the string has the suffix `Lt` it should have length >= 2. I think this should be okay since we only slice the end off the keys if they have the corresponding suffixes.